scripts/template_docsting_with_example.py

```diff
@@ -0,0 +1,77 @@
+# see more documentation guidelines online here: https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP)
+
+
+class SomeClass:
+    """A one line description of the Class.
+
+    A short paragraph providing more details.
+
+    Deprecation warning
+
+    Extended Summary
+
+    Parameters
+    ----------
+    scale_factor :class:`float`
+        The factor used for scaling.
+
+    Returns
+    -------
+    :class:`VMobject`
+        Returns self.
+
+    Tests
+    -----
+
+    Yields
+    -------
+
+    Receives
+    ----------
+
+    Other Parameters
+    -----------------
+
+    Raises
+    ------
+    :class:`TypeError`
+        If one element of the list is not an instance of VMobject
+
+    Warns
+    -----
+
+    Warnings
+    --------
+
+    Notes
+    -----
+
+    Examples
+    --------
+    .. manim:: AddTextLetterByLetterScene
+        :save_last_frame:
+
+        class AddTextLetterByLetterScene(Scene):
+            def construct(self):
+                t = Text("Hello World word by word")
+                self.play(AddTextWordByWord(t))
+
+
+
+    See Also
+    --------
+    :class:`ShowCreation`, :class:`~.ShowPassingFlash`
+
+    References
+    ------
```

Comment:
The underline always has to be at least as long as the caption, otherwise Sphinx generates a new warning.

```suggestion
    References
    ----------
```

Reply:
Thanks!